src/include/sof/common.h

```diff
@@ -11,6 +11,10 @@
 
 #if !defined(LINKER)
 
+#ifndef typeof
+#define typeof  __typeof__
+#endif
+
```

Comment:
Don't we have a Zephyr solution for this? We should be shrinking, not growing the sof/common.h.

Reply:
we could just replace all `typeof()` with `__typeof__()`, I haven't found anything *global* in Zephyr, they have two "local" definitions in arch/arm/include/cortex_m/cmse.h and in drivers/usb/device/usb_dc_rpi_pico.c

Reply:
@kv2019i I think I'll do that - replace that patch with a switch to `__typeof__()` globally in SOF